src/ducks/apps/components/Sections/Sections.jsx

```diff
@@ -108,110 +109,96 @@ export class Sections extends Component {
       ]
     }
 
-    const { apps } = this.props
+    const { filteredApps } = this.state
 
-    const appsForSearch = apps.map(app => ({
+    const appsForSearch = filteredApps.map(app => ({
       ...app,
       doctypes: Object.values(app.permissions).map(x => x.type)
     }))
     this.fuse = new Fuse(appsForSearch, options)
   }
 
-  /** Rename as filter */
-  getSearch() {
-    // Depending on whether the component is controlled or uncontrolled,
-    // search is taken from props or state
-    const search = this.props.search || this.state.search
-    return search
+  componentDidUpdate(prevProps) {
+    if (
+      prevProps.filter !== this.props.filter ||
+      prevProps.apps !== this.props.apps
+    ) {
+      this.updateFilteredApps()
+      this.setState({ searchResults: null, searchFieldValue: '' })
```

Comment:
doesn't that mean you remove the value of the searchField input when a search is done? 

I have difficulty understanding this part

Reply:
The apps are coming from the above, the filter is done on the categories of apps, and is modified for example when switching categories either by going to a different location via the nav or by selecting another category option. I think I'll add a docstring explaining the differnce between the filter (from the props), and the search (from the state) in the docstring of Section to make it clearer. What do you think ? Does it make more sense to you ?

Reply:
Done in https://github.com/cozy/cozy-store/pull/694/commits/e2e493b4355d89dff90020daca01c63836e064fb